cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -362,9 +363,10 @@ findProjectRoot = do
 readProjectConfig :: Verbosity -> FilePath -> Rebuild ProjectConfig
 readProjectConfig verbosity projectRootDir = do
     global <- readGlobalConfig verbosity
-    local  <- readProjectLocalConfig      verbosity projectRootDir
-    extra  <- readProjectLocalExtraConfig verbosity projectRootDir
-    return (global <> local <> extra)
+    local  <- readProjectLocalConfig       verbosity projectRootDir
+    freeze <- readProjectLocalFreezeConfig verbosity projectRootDir
```

Comment:
So here's my theory that tries to unify what is otherwise an ad-hoc list of config files: the `cabal.project` file would contain a `include cabal.project.*` directive, and certain commands like configure, freeze etc manage files that can be included. By default they would be included but we'd make that all completely explicit.

So in future, a `cabal project init` might make a default `cabal.project` file like:
```
include $global/config
include cabal.project.*
packages: ./
optional-packages: ./*/
```

Having them as separate files resolves the problem of automatically editing manually managed files. It's not that it's technically a problem to edit the main `cabal.project` file, e.g. we could have commands to help do that like adding a new package into the list, but the problem that we don't know how to edit things because we cannot distinguish between manually specified things and generated things.